### ISSUE TITLE
Monitoring - project monitoring does not work in upgraded setup

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#22862]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- run rancher v2.2.8 in airgap HA mode with external system-chart 
- add cluster1 and cluster2
- for both clusters, enable cluster monitoring and project monitoring in one project 
- upgrade Rancher to v2.3.0-rc3
- there is no new version for monitoring since the system chart still use branch `release-v2.2`
- edit the system-catalog to use branch `v2.3-dev`
- now see new version available in the UI 
- in cluster1, upgrade the cluster monitoring, then upgrade the project monitoring
- deploy a workload with metrics exposed, image `loganhz/web`

**Bug 1**: the project Prometheus cannot detect the workload as a target, thus we cannot get search the expression `web_app_online_user_count` in Prometheus 

- in cluster1 -> project 2, enable the project monitoring

**Bug 2**: the project monitoring app deploys only one workload, the second `prometheus-project-monitoring` is not deployed, thus we cannot open the project Prometheus page

<img width="1429" alt="Screen Shot 2019-09-13 at 4 00 26 PM" src="https://user-images.githubusercontent.com/6218999/64899314-9e61bd00-d63f-11e9-8ab1-35b1842086ee.png">
<img width="494" alt="Screen Shot 2019-09-13 at 4 02 29 PM" src="https://user-images.githubusercontent.com/6218999/64899377-e7b20c80-d63f-11e9-8208-24608a9d871c.png">

**Bug 3**: For the cluster monitoring page and project monitoring page, in cluster1 it shows two versions: `0.0.3` and `0.0.4`, but in cluster2 it shows only `0.0.4` This is also tracked here: https://github.com/rancher/rancher/issues/22842#issuecomment-531390968

**Bug 4**: in cluster2, cannot upgrade cluster monitoring to `0.0.4` since the API POST request does not send the version. 

This screenshot is a vliad API request: 

<img width="766" alt="Screen Shot 2019-09-13 at 3 23 20 PM" src="https://user-images.githubusercontent.com/6218999/64899529-b4bc4880-d640-11e9-81ca-50272fd8be47.png">

**Comment by loganhz** [comment]
@jiaqiluo 

Cannot see Bug 2 any more in your setup. 
Bug 3 and Bug 4 already fixed in master-head


@orangedeng please look at issue one. I didn't see the prometheus config updated with custom metrics. So there is no server discovery and targets entry to the custom metrics.

**Comment by loganhz** [comment]
```
E0914 02:59:39.410560 1 reflector.go:125] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:450: Failed to list *v1.PodMonitor: the server could not find the requested resource (get podmonitors.monitoring.coreos.com)
```

Error from operator.

It looks like the newly added CRD is not installed.

@guangbochen It looks like we will not install the CRD if we upgrade operator and monitoring app in apps page

**Comment by cjellick** [comment]
will have to wait for an RC to test

**Comment by jiaqiluo** [comment]
The bug fix is validated in the upgrade path v2.2.8 -> v2.3.0-rc4, single-node install 

Steps are the same as the issue describes. 

Validation for bug 1: after upgrading the project monitoring, it can still detect new metrics 
Validation for bug 2: enabling project monitoring in a new project, the project monitoring app deploys two workloads and works as expected
Validation for bug 3 and 4: see validation at https://github.com/rancher/rancher/issues/22842
